contrib/federations_tutorial/MultiSig.py

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/env python3
+import os
+import random
+import sys
+import time
+import shutil
+import subprocess
+from util import *
+from authproxy import AuthServiceProxy, JSONRPCException
+
+class MultiSig():
+    def __init__(self, nodes, sigs, elementsdir):
+        self.num_of_nodes = nodes
+        self.num_of_sigs = sigs
+        self.elements_dir = elementsdir
+        self.privkeys = []
+        self.multisig = ""
+
+    def generate(self):
+        pubkeys = []
+        for i in range(0, self.num_of_nodes): # get new key from each node
+            temp_datadir = "/tmp/"+''.join(random.choice('0123456789ABCDEF') for i in range(5))
```

Comment:
You can reuse the same tmp dir since then you have node_i in the path for each node.
Perhaps some things could be simplified reusing https://github.com/ElementsProject/elements/blob/elements-0.14.1/qa/rpc-tests/test_framework/util.py#L337